bin/faucet/src/main.rs

```diff
@@ -72,6 +73,26 @@ async fn main() -> Result<(), FaucetError> {
 
             info!(target: COMPONENT, %config, "Initializing server");
 
+            let client_state_clone = Arc::clone(&faucet_state.client);
+
+            info!("Initializing chain tip updater");
+            thread::spawn(move || {
+                let sys = actix_web::rt::System::new();
```

Comment:
Nit: consider moving the clone inside the spawn

```suggestion
            let client_state_clone = Arc::clone(&faucet_state.client);

            info!("Initializing chain tip updater");
            thread::spawn({
                let client_state = Arc::clone(&faucet_state.client);
                move || {
                    let sys = actix_web::rt::System::new();
```

Reply:
I can't move the close inside the spawn since it will take ownership of the `faucet_state`

Reply:
Does it? Hmm. I thought maybe it wouldn't because it comes before the `move ||`.